vumi/transports/smpp/smpp_transport.py

```diff
@@ -402,6 +406,8 @@ def handle_delivery_report(self, receipted_message_id, delivery_status):
         dr = yield self.publish_delivery_report(
             user_message_id=message_id,
             delivery_status=delivery_status)
+
+        yield self.message_stash.delete_remote_message_id(receipted_message_id)
```

Comment:
I think there is a problem here -- if we delete the message when the delivery report status is something other than delivered, we won't be able to find the message if it does get delivered. Maybe we should only delete the remote message if the delivery report indicates complete failure or successful delivery?